arc-mlir/src/include/arc/Ops.td

```diff
@@ -258,6 +261,62 @@ def IfOp : Arc_Op<"if", [SingleBlockImplicitTerminator<"ArcBlockResultOp">]> {
   let hasCanonicalizer = 1;
 }
 
+def ArcYieldOp : Arc_Op<"yield", [Terminator, SameOperandsAndResultType]> {
```

Comment:
This appears to almost the same as the `ArcBlockResultOp`, do we need both? `ArcYieldOp` is a better name so I prefer to keep the new operator. Looking at `ArcBlockResultOp` I see that it doesn't have any parent constraint, should you not require it to be a child of a loop? I think it should be possible to an 'or' constraint and use it both for if and loops.

Reply:
I'll try it out